R/gam.R

```diff
@@ -0,0 +1,106 @@
+#' Axing a gam.
+#'
+#' gam objects are created from the \pkg{mgcv} package.
+#'
+#' @inheritParams butcher
+#'
+#' @return Axed gam object.
+#'
+#' @examplesIf rlang::is_installed("mgcv")
+#'
+#' cars_gam <- mgcv::gam(mpg ~ s(disp, k = 3) + s(wt), data = mtcars)
+#' cleaned_gam <- butcher(cars_gam, verbose = TRUE)
+#'
+#' @name axe-gam
+NULL
+
+#' Remove the call.
+#'
+#' @rdname axe-gam
+#' @export
+axe_call.gam <- function(x, verbose = FALSE, ...) {
+  old <- x
+  x <- exchange(x, "call", call("dummy_call"))
+  x <- exchange(x, "mgcv.conv.score.calls", call("dummy_call"))
```

Comment:
I don't see this exact thing in a simple example, does it come up in other places?

``` r
cars_gam <- mgcv::gam(mpg ~ s(disp, k = 3) + s(wt), data = mtcars)

names(cars_gam$mgcv.conv)
#> [1] "full.rank"       "rank"            "fully.converged" "hess.pos.def"   
#> [5] "iter"            "score.calls"     "rms.grad"

cars_gam$mgcv.conv$score.calls
#> [1] 7
```

<sup>Created on 2022-08-19 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Reply:
Thank you for this! I was convinced there was something else in there because of the output of `locate(cars_gam, "call")` but there is no call here.